ca_tools.erl

```diff
@@ -1,40 +1,42 @@
 -module(ca_tools).
--export([iterate/3, make_rule/1, make_state/1, random_state/1]).
+-export([start/4]).
 
--spec iterate(ca:state(), [ca:rule()], non_neg_integer()) -> ca:state().
-iterate(State, _, 0) ->
+-type automaton() :: elementary | von_neumann.
+-type state_type() :: random | origin.
+
+-spec start(automaton(), {state_type(), integer()}, integer(), non_neg_integer()) -> ca:state().
+start(AutomatonType, {StateType, StateSize}, RuleNumber, Iterations) ->
+    CaModule = case AutomatonType of 
+                   elementary -> ca;
+                   von_neumann -> ca2d
+               end,
+    {{bits, Bits}, {make_state, StateGenerator}} = CaModule:get_start(),
+    State = StateGenerator(StateType, StateSize),
+    Rules = make_rule(RuleNumber, Bits),
+    iterate(CaModule, State, Rules, Iterations).
+
+
+-spec iterate(module(), ca:state(), [ca:rule()], non_neg_integer()) -> ca:state().
```

Comment:
Line 19 is too long: -spec iterate(module(), ca:state(), [ca:rule()], non_neg_integer()) -> ca:state()..